generators/app/tasks/installDependencies.js

```diff
@@ -53,9 +82,15 @@ module.exports = function installDependencies() {
   if (this.features.drawerios || this.features.drawerandroid) {
     DEPENDENCIES.push('react-native-drawer');
   }
-  return yarnInstall(this.projectName, DEPENDENCIES, this.options)
-    .then(() => yarnInstall(this.projectName, DEV_DEPENDENCIES, this.options, true))
-    .catch(() => {
-      process.exit(1);
-    });
+  return getLinterPluginVersions(this.projectName, this.options).then(plugins => {
+    const pluginNames = Object.keys(plugins);
+    const fixedDevDeps = DEV_DEPENDENCIES.map(
+      dependency => (pluginNames.includes(dependency) ? `${dependency}@${plugins[dependency]}` : dependency)
+    );
+    return yarnInstall(this.projectName, DEPENDENCIES, this.options)
+      .then(() => yarnInstall(this.projectName, fixedDevDeps, this.options, true))
+      .catch(() => {
+        process.exit(1);
```

Comment:
I'd keep this `1` as is, but if we are going to add more exit codes later, it is desirable to extract this into a constants file.